qa/workunits/rbd/cli_generic.sh

```diff
@@ -1463,6 +1463,174 @@ test_mirror_snapshot_schedule_recovery() {
+    rbd mirror pool enable rbd2/ns1 image
+    rbd mirror pool peer add rbd2 cluster1
+
+    test "$(ceph rbd mirror group snapshot schedule list)" = "{}"
+    ceph rbd mirror group snapshot schedule status | fgrep '"scheduled_groups": []'
+
+    expect_fail rbd mirror group snapshot schedule ls
+    test "$(rbd mirror group snapshot schedule ls -R --format json)" = "[]"
+
+    rbd create $RBD_CREATE_ARGS -s 1 rbd2/ns1/img1
+    rbd create $RBD_CREATE_ARGS -s 1 rbd2/ns1/img2
+    rbd group create rbd2/ns1/gp1
+    rbd group image add rbd2/ns1/gp1 rbd2/ns1/img1
+    rbd group image add rbd2/ns1/gp1 rbd2/ns1/img2
+
+    test "$(rbd group snap ls rbd2/ns1/gp1 | grep -c mirror.primary)" = '0'
+
+    rbd mirror group enable rbd2/ns1/gp1 snapshot
+
+    test "$(rbd group snap ls rbd2/ns1/gp1 | grep -c mirror.primary)" = '1'
+
+    rbd mirror group snapshot schedule add -p rbd2/ns1 --group gp1 1m
+    expect_fail rbd mirror group snapshot schedule ls
+    rbd mirror group snapshot schedule ls -R | grep 'rbd2 *ns1 *gp1 *every 1m'
+    expect_fail rbd mirror group snapshot schedule ls -p rbd2
+    rbd mirror group snapshot schedule ls -p rbd2 -R | grep 'rbd2 *ns1 *gp1 *every 1m'
+    expect_fail rbd mirror group snapshot schedule ls -p rbd2/ns1
+    rbd mirror group snapshot schedule ls -p rbd2/ns1 -R | grep 'rbd2 *ns1 *gp1 *every 1m'
+    test "$(rbd mirror group snapshot schedule ls -p rbd2/ns1 --group gp1)" = 'every 1m'
+
+    for i in `seq 12`; do
+        test "$(rbd group snap ls rbd2/ns1/gp1 | grep -c mirror.primary)" -gt '1' && break
+        sleep 10
+    done
+
+    test "$(rbd group snap ls rbd2/ns1/gp1 | grep -c mirror.primary)" -gt '1'
+
+    # repeat with kicked in schedule, see https://tracker.ceph.com/issues/53915
+    expect_fail rbd mirror group snapshot schedule ls
+    rbd mirror group snapshot schedule ls -R | grep 'rbd2 *ns1 *gp1 *every 1m'
+    expect_fail rbd mirror group snapshot schedule ls -p rbd2
+    rbd mirror group snapshot schedule ls -p rbd2 -R | grep 'rbd2 *ns1 *gp1 *every 1m'
+    expect_fail rbd mirror group snapshot schedule ls -p rbd2/ns1
+    rbd mirror group snapshot schedule ls -p rbd2/ns1 -R | grep 'rbd2 *ns1 *gp1 *every 1m'
+    test "$(rbd mirror group snapshot schedule ls -p rbd2/ns1 --group gp1)" = 'every 1m'
+
+    rbd mirror group snapshot schedule status
+    test "$(rbd mirror group snapshot schedule status --format xml |
+        xmlstarlet sel -t -v '//scheduled_groups/group/group')" = 'rbd2/ns1/gp1'
+    test "$(rbd mirror group snapshot schedule status -p rbd2 --format xml |
+        xmlstarlet sel -t -v '//scheduled_groups/group/group')" = 'rbd2/ns1/gp1'
+    test "$(rbd mirror group snapshot schedule status -p rbd2/ns1 --format xml |
+        xmlstarlet sel -t -v '//scheduled_groups/group/group')" = 'rbd2/ns1/gp1'
+    test "$(rbd mirror group snapshot schedule status -p rbd2/ns1 --group gp1 --format xml |
+        xmlstarlet sel -t -v '//scheduled_groups/group/group')" = 'rbd2/ns1/gp1'
+
```

Comment:
I'd expect the image snapshot schedule to be empty at this point . Please add a test to check it as well.

Reply:
Done